aiowebostv/webos_client.py

```diff
@@ -112,65 +111,72 @@ def registration_msg(self) -> dict[str, Any]:
         handshake["payload"]["client-key"] = self.client_key  # type: ignore[index]
         return handshake
 
-    async def _ws_connect(
-        self, uri: str, ssl_context: ssl.SSLContext | None
-    ) -> WebSocketClientProtocol:
+    async def _ws_connect(self, uri: str) -> ClientWebSocketResponse:
         """Create websocket connection."""
         _LOGGER.debug("connect(%s): uri: %s", self.host, uri)
-        return await ws_connect(
-            uri,
-            ping_interval=self.ping_interval,
-            ping_timeout=self.ping_timeout,
-            open_timeout=self.timeout_connect,
-            close_timeout=self.timeout_connect,
-            max_size=None,
-            ssl=ssl_context,
-        )
+
+        if TYPE_CHECKING:
+            assert self.client_session is not None
+
+        # webOS uses self-signed certificates, disable SSL certificate validation
+        async with asyncio.timeout(self.timeout_connect):
+            return await self.client_session.ws_connect(
+                uri, heartbeat=self.heartbeat, ssl=False
+            )
+
+    async def close_client_session(self) -> None:
+        """Close client session if we created it."""
+        if TYPE_CHECKING:
+            assert self.client_session is not None
+
+        await self.client_session.close()
+        self.created_client_session = False
+        self.client_session = None
 
     async def connect_handler(self, res: Future) -> None:
         """Handle connection for webOS TV."""
         handler_tasks: set[Task] = set()
-        main_ws: WebSocketClientProtocol | None = None
-        input_ws: WebSocketClientProtocol | None = None
-        ssl_context: ssl.SSLContext | None = None
+        main_ws: ClientWebSocketResponse | None = None
+        input_ws: ClientWebSocketResponse | None = None
+
         try:
+            # Create a new client session if not provided
+            if self.client_session is None:
+                self.client_session = ClientSession()
+                self.created_client_session = True
+
```

Comment:
Does this need to be in the `try`?

Reply:
Until I refactor this block, the `try` is used to make sure we go to the `finally:`
https://github.com/home-assistant-libs/aiowebostv/blob/a0387e4bc32c7f78feabe89f6a0de009b087389a/aiowebostv/webos_client.py#L252
Which creates the cleanup task to close the session:
https://github.com/home-assistant-libs/aiowebostv/blob/a0387e4bc32c7f78feabe89f6a0de009b087389a/aiowebostv/webos_client.py#L267-L268